src/ClientData/include/ClientData/ScopeIdProvider.h

```diff
@@ -107,8 +108,32 @@ class NameEqualityScopeIdProvider : public ScopeIdProvider {
   /// TODO(http://b/247467504): Add FunctionInfo to ScopeInfo.
   absl::flat_hash_map<ScopeId, FunctionInfo> scope_id_to_function_info_;
   mutable absl::Mutex mutex_;
-};
 
+  // The code below is an optimized version of the following
+  // ```
+  // static ScopeType ScopeTypeFromTimerInfo(const TimerInfo& timer) {
+  //  switch (timer.type()) {
+  //    case TimerInfo::kNone:
+  //      return timer.function_id() != orbit_grpc_protos::kInvalidFunctionId
```

Comment:
Since you modified the code, consider switching the if statement:

```
//      return timer.function_id() == orbit_grpc_protos::kInvalidFunctionId
//                 ? ScopeType::kInvalid;
//                 : ScopeType::kDynamicallyInstrumentedFunction
```